### PR TITLE
Remove Python 3.10 support for the controller

### LINKED ISSUE #83094: Drop support for Python 3.10 on the controller
### Summary

Drop support for Python 3.10 on the controller.

### Issue Type

Feature Idea

### Component Name

`ansible-test`

### PR DESCRIPTION
**Pull request by mkrizek**
##### SUMMARY
Fixes #83094

##### ISSUE TYPE
- Feature Pull Request

### CODE REVIEW

File: lib/ansible/utils/collection_loader/_collection_finder.py
Comment (mattclay) [inline lib/ansible/utils/collection_loader/_collection_finder.py:35]: This Python module needs to support target Python versions, not just controller Python versions. So the Python 2.7 support can be dropped, but not Python 3.8.

Comment (mattclay) [reply lib/ansible/utils/collection_loader/_collection_finder.py:35]: On a related note, there is more Python 2.7 code that can be removed, such as the use of `ansible.module_utils.six`.

**Comment by mattclay** [comment]
This PR is blocked on completion of https://github.com/ansible/ansible/issues/83093

**Comment by mattclay** [comment]
I've restarted the tests now that the AZP container has been updated to include Python 3.11.

Before this PR is merged we should check on the status of the updates to the other projects using AZP: https://github.com/ansible-collections/news-for-maintainers/issues/71

Any projects testing against `devel` will start failing once this is merged, if they haven't updated to the new container.

File: lib/ansible/galaxy/collection/__init__.py
Comment (mattclay) [inline lib/ansible/galaxy/collection/__init__.py:1608]: There are unit tests that still refer to this and will need to be updated.

File: lib/ansible/utils/collection_loader/_collection_finder.py
Comment (felixfontein) [reply lib/ansible/utils/collection_loader/_collection_finder.py:35]: `ansible.module_utils.six` is used in quite a few collections all over the place.

Comment (mattclay) [reply lib/ansible/utils/collection_loader/_collection_finder.py:35]: We're not removing that feature -- we're just discontinuing use of it in core where it's no longer needed.

Comment (felixfontein) [reply lib/ansible/utils/collection_loader/_collection_finder.py:35]: That's totally sensible :) Thanks for clarifying!

**Comment by mattclay** [comment]
@mkrizek Is there anything left to do on this, other than rebase the PR?

**Review by mattclay — APPROVED** [review]

**Pull request merged**
